tokio-util/src/future/with_cancellation_token.rs

```diff
@@ -0,0 +1,73 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use pin_project_lite::pin_project;
+
+use crate::sync::{CancellationToken, RunUntilCancelledFuture, RunUntilCancelledFutureOwned};
+
+pin_project! {
+    /// A Future that is resolved once the corresponding [`CancellationToken`]
+    /// is cancelled or a given Future gets resolved. It is biased towards the
+    /// [`CancellationToken`] cancelled.
+    #[must_use = "futures do nothing unless polled"]
+    pub struct WithCancellationTokenFuture<'a, F: Future> {
+        #[pin]
+        run_until_cancelled: Option<RunUntilCancelledFuture<'a, F>>
+    }
+}
```

Comment:
```suggestion
pin_project! {
    /// A [`Future`] that is resolved once the corresponding [`CancellationToken`]
    /// is cancelled or a given [`Future`] gets resolved.
    ///
    /// This future is Immediately resolved if the corresponding [`CancellationToken`]
    /// is already cancelled, otherwise, in case of concurrent completion and
    /// cancellation, this is biased towards the future completion.
    #[must_use = "futures do nothing unless polled"]
    pub struct WithCancellationTokenFuture<'a, F: Future> {
        #[pin]
        run_until_cancelled: Option<RunUntilCancelledFuture<'a, F>>
    }
}
```

> It is biased towards the [`CancellationToken`] cancelled.

Emm, my understanding of this sentence is that this future is always biased towards the cancellation, but this is only true if the token is already cancelled before constructing this struct.

So I tried to write another version to avoid confusion.